subiquity/client/controllers/ubuntu_pro.py

```diff
@@ -85,6 +86,12 @@ def cancel(self) -> None:
         self.app.prev_screen()
 
     def done(self, token: str) -> None:
+        """ Submit the token and move on to the next screen. """
         self.app.next_screen(
             self.endpoint.POST(UbuntuProInfo(token=token))
         )
+
+    def next_screen(self) -> None:
```

Comment:
This is a bit unusual as the client controllers go.  Can you elaborate?  Did this used to have an extra step before app.next_screen()?

Reply:
So, the reason why I added this is because I wanted to submit the contract token to the server _before_ showing the "Token added successfully" popup (and also before showing the screen with the list of services)

so we:
1. submit the token through POST /ubuntu_pro/ (this marks the controller configured)
2. display the "token added successfully" popup
3. display the screen with the list of services

if we implemented step 1. with a simple call to `controller.done()` (as most sets of screens do), we would directly move on to the SSH screen ; without giving us a chance to do step 2. and 3.

in the previous implementation, we would wait until the user clicks OK in the list of services popup to submit the token:

![ok](https://user-images.githubusercontent.com/4038023/175275419-6a58491d-b865-42e8-8374-706b1e1e78d6.png)

But this popup had no "Back" button back then.
Now, we want the settings to be saved even if the users decides to go back ; so that they don't start over with an empty token.

![back](https://user-images.githubusercontent.com/4038023/175275864-40318c5f-28a2-499f-908b-16d64ae5538e.png)

Let me know if that does not make sense. I can show examples.